src/home/fetch-github/fetch-issues-preview.ts

```diff
@@ -1,18 +1,69 @@
 import { Octokit } from "@octokit/rest";
-import { getGitHubAccessToken } from "../getters/get-github-access-token";
+import { getGitHubAccessToken, getGitHubUserName } from "../getters/get-github-access-token";
 import { GitHubIssue } from "../github-types";
 import { taskManager } from "../home";
 import { displayPopupMessage } from "../rendering/display-popup-modal";
 import { TaskNoFull } from "./preview-to-full-mapping";
 
+async function checkPrivateRepoAccess(): Promise<boolean> {
+  const octokit = new Octokit({ auth: await getGitHubAccessToken() });
+  const username = getGitHubUserName();
+
+  try {
+    const response = await octokit.repos.checkCollaborator({
+      owner: "ubiquity",
+      repo: "devpool-directory-private",
+      username,
+    });
+
+    if (response.status === 204) {
+      // If the response is successful, it means the user has access to the private repository
+      return true;
+    }
+    return false;
+  } catch (error) {
+    if (error.status === 404) {
+      // If the status is 404, it means the user is not a collaborator, hence no access
+      return false;
+    } else {
+      // Handle other errors if needed
+      console.error("Error checking repository access:", error);
+      throw error;
+    }
+  }
+}
+
 export async function fetchIssuePreviews(): Promise<TaskNoFull[]> {
-  const octokit = new Octokit({ auth: getGitHubAccessToken() });
+  const octokit = new Octokit({ auth: await getGitHubAccessToken() });
 
   let freshIssues: GitHubIssue[] = [];
+  let hasPrivateRepoAccess = false; // Flag to track access to the private repository
+
   try {
-    const response = await octokit.paginate<GitHubIssue>("GET /repos/ubiquity/devpool-directory/issues", { state: "open" });
+    // Check if the user has access to the private repository
+    hasPrivateRepoAccess = await checkPrivateRepoAccess();
+
+    // Fetch issues from public repository
+    const publicResponse = await octokit.paginate<GitHubIssue>("GET /repos/ubiquity/devpool-directory/issues", { state: "open" });
```

Comment:
You should be using built in methods. Something like the following:


```typescript
const issues = await octokit.issues.listForRepo({
  owner: "ubiquity",
  repo: "devpool-directory",
  state: "open"
});
```

Reply:
I'd switch it but i saw it used that way, don't know if there is a reason